### ISSUE TITLE
Permutation test for cellProximityEnrichment(), including only heterotypic interactions

### ISSUE THREAD

**Issue by evanmaes** [giotto-suite/Giotto#208]
Hi all, 

TLDR: Creating an option for null distribution based on only heterotypic interactions

I am using your cellProximityEnrichment() function to test if certain cell-cell interactions are enriched in my CODEX data. I am using Giotto version 1.0.4. For the purposes of my analysis I would like to consider only heterotypic interactions (and exclude homotypic interactions) when generating a simulated network and calculating the p-values. Do you have any suggestions on how to implement this?

Because you label the edges of the spatial network as homo- or hetero-typic, I feel like it should be easy to modify for this functionality. I tried modifying the [make_simulated_network()](https://github.com/RubD/Giotto/blob/master/R/spatial_interaction.R) function.

```
#remove homo interactions (added to line 30)
spatial_network_annot=spatial_network_annot[type_int=='hetero']

#Select only hetero-interactions (added to line 69)
sample_dt=sample_dt[type_int=='hetero']

#select only hetero interactions to calculate p-val based on (added to line 176)
table_results<-table_results[type_int=='hetero']

```

However, I realized despite reducing the random sampling pool to include only the hetero-interactions, the total # of interactions in the actual original network and simulated network will not be equal with this strategy. Because your method randomly samples and create 2 new lists to construct the background null distribution there would always be homo-homo interactions across a row. 
![image](https://user-images.githubusercontent.com/47531107/160514646-138c06ac-3c3d-4e82-bf89-f9717dceea65.png)

Ultimately what I think I would need is 2 randomly sampled lists, without duplications across a row (preventing a homotypic interaction). This would maintain the total # of simulated interactions equaling the simulated network, and only include heterotypic interactions for later p-value calculation. This would just require modifying s1_list and s2_list in the lines of make_simulated_network() below. But, I'm having a difficult time thinking about how to do this step.

```
reshuffled_all_cell_type = sample(x = all_cell_type, size = length(all_cell_type), replace = F)

new_from_cell_type = reshuffled_all_cell_type[1:middle_point]
s1_list[[sim]] = new_from_cell_type

new_to_cell_type = reshuffled_all_cell_type[(middle_point+1):length(all_cell_type)]
s2_list[[sim]] = new_to_cell_type
```

Any advice to implement this would be greatly appreciated! For some reason, the homo-homo interactions were dominating the p-values in all of my tumor samples when running your simulations with default settings. I am interested more in the potential enrichment of hetero interactions like T cell--Tumor or Macrophage--Tumor, etc (and less so Tumor--Tumor or Macrophage--Macrophage).

Thanks again for your wonderful package :)

**Comment by RubD** [comment]
This sounds like something we could do, but not sure how dangerous this could be? Since excluding the homo-typic interactions would in fact mean that you distort the original spatial architecture, which is what you need to create the null distribution. We also see that homo-typic interactions are typically found to be the top enriched pairwise interactions in most tissues, but it's still possible to find statistically enriched hetero-typic interactions. 

I guess we could limit the null distribution to the initial subset of heterotypic interactions within the spatial network. This would still create homotypic interactions within this subset, but at least it wouldn't be dominated by the original large number of homo-typic interactions.
